chore(blog): add May community spotlight (#4111)
asyncapi/website: markdown/blog/community-spotlight-aayush.md

---
title: "Inside the Open Source Climb: Aayush’s Path to AsyncAPI Leadership"

> date: 2025-04-28T06:00:00+01:00

Reviewer: @iambami This is something you need to change now.

tags:
  - Project Status
  - Community Spotlight
cover: 
authors:
  - name: Atinuke Bami Kayode
    photo: /img/avatars/bami.webp
    link: https://www.linkedin.com/in/atinuke-Bami-kayode-5b838b1b7/
    byline: AsyncAPI Community Marketing Specialist
excerpt: "When Aayush made his first contribution to AsyncAPI, he didn’t expect it would lead to mentorship, leadership, and a thriving open source career."
---

Open-source communities don’t just grow with code, they grow with people. Every successful project is powered by individuals whose passion, persistence, and commitment light the way for others.
In this month’s **Community Spotlight,** we’re thrilled to feature [Aayush](https://www.linkedin.com/in/aayush-saini-0a25931b1/), a shining example of how consistent contributions and a willingness to learn can evolve into leadership. From making his first pull request to mentoring others through Google Summer of Code, Aayush’s story is a powerful reminder of what’s possible when a supportive community and personal drive meet.
In this interview, Aayush shares his journey, challenges, growth, and advice for new contributors stepping into the AsyncAPI world.

## The Interview

> **Bami: Nice to meet you, Aayush! I saw the map of your journey, it’s super interesting. Your name was even nominated by [Thulie](https://www.linkedin.com/in/v-thulisile-sibanda/]) initially. Thanks for making it easy to reach out. Shall we get started?**

Reviewer 2: Here's what Coderabbit meant @iambami 
```suggestion
**Bami: Nice to meet you, Aayush! I saw the map of your journey, it’s super interesting. Your name was even nominated by [Thulie](https://www.linkedin.com/in/v-thulisile-sibanda/)  initially. Thanks for making it easy to reach out. Shall we get started?**
```

**You didn’t start with Google Summer of Code, right? You started as a regular contributor?**

> ***Aayush:** Yeah, I started as a normal contributor. After contributing to the CLI project, I applied for the SNKP mentorship program. After completing that mentorship, I became a maintainer and then a TSC member. Currently, I'm mentoring two projects in Google Summer of Code.*

Reviewer 3: Applied for Asyncapi Mentorship

**Bami: That's amazing! Could you give us a quick background about yourself, where you're from, your interests, and your journey so far?**

> ***Aayush:** Sure! My name is Aayush. I'm currently working as a Software Engineer at Reddit. I’ve been contributing to open source for the past three years across organizations like Textile and Jenkins. I got into AsyncAPI because I wanted to learn about event-driven architecture and TypeScript. When I first started, the ecosystem was huge and a bit overwhelming. I decided to begin with the CLI project, setting it up locally and gradually picking up small issues. I kept solving problems, creating pull requests, and helping others with conflicts. Eventually, I built my confidence and expanded my contributions, including winning bounties and developing workflows that are still in use today.*

Reviewer 3: Red Hat instead of Reddit

Reviewer 3: Like Checkstyle not textile : )

Author: thank you

> ***Aayush:** Sure! My name is Aayush. I'm currently working as a Software Engineer at Reddit. I’ve been contributing to open source for the past three years across organizations like Textile and Jenkins. I got into AsyncAPI because I wanted to learn about event-driven architecture and TypeScript. When I first started, the ecosystem was huge and a bit overwhelming. I decided to begin with the CLI project, setting it up locally and gradually picking up small issues. I kept solving problems, creating pull requests, and helping others with conflicts. Eventually, I built my confidence and expanded my contributions, including winning bounties and developing workflows that are still in use today.*

Reviewer 2: [suggestion]: To add onto Aayush's advice, it would be helpful to hyperlink the names of the organizations mentioned, so that people who are interested in learning more about them can easily access their content. 
```suggestion
***Aayush:** Sure! My name is Aayush. I'm currently working as a Software Engineer at [Red Hat](https://www.redhat.com/en). I’ve been contributing to open source for the past three years across organizations like [Checkstyle](https://checkstyle.org/) and [Jenkins](https://www.jenkins.io/). I got into AsyncAPI because I wanted to learn about event-driven architecture and TypeScript. When I first started, the ecosystem was huge and a bit overwhelming. I decided to begin with the CLI project, setting it up locally and gradually picking up small issues. I kept solving problems, creating pull requests, and helping others with conflicts. Eventually, I built my confidence and expanded my contributions, including winning bounties and developing workflows that are still in use today.*
```

***Aayush:** Haha, yes!*

**Bami: Have you ever been surprised by how fast someone reviewed your PR?**
***Aayush:** Yes, [Lukasz](https://www.linkedin.com/in/lukasz-gornicki-a621914/) once reviewed my PR very quickly and I was amazed.*

**Bami: Have you ever learned something new from a fellow AsyncAPI contributor?**
***Aayush:** Yes, I learned a lot from Ashish, Souvik, and others.*

## What’s Next for Aayush?

As a TSC member, mentor, maintainer, and active community contributor, Aayush has exciting goals:

- Take ownership of the Studio project.

- Continue working on the Java + Quarkus onboarding template.

- Explore and contribute more to frontend technologies.

- Keep enhancing tools like the Git voting automation system, a project he worked on during a bounty period, pushing over 400 commits to test and perfect it!

> *"There’s a lot of fun stuff going on, and I’m excited for what's ahead,"* he said.

Bami: Awesome, you’ve been an amazing guest. Thank you so much for sharing your story with the rest of the community, Aayush!

It was such a pleasure speaking with Aayush and hearing his journey, full of perseverance, community spirit, and continuous growth.

> If you're looking to start your journey in open source, his story is a great reminder: **take your time, be patient, stay curious, and support others along the way.**

Reviewer 2: [suggestion]: The ending reinforces the theme of Aayush's story but it could use a call to action. Telling people to take your time is great but there might be folks who want to immediately start their open source journey and encouraging them to join AsyncAPI's community would be a great way to convey this point. 

```suggestion
Aayush's story is a great reminder to: **take your time, be patient, stay curious, and support others along the way.** If you're looking to start your own journey in open source, join our Slack community and check out our repository! We look forward to seeing your contributions.
```

---
title: "Inside the Open Source Climb: Aayush’s Path to AsyncAPI Leadership"

> date: 2025-06-07T06:00:00+01:00

Author: We are leaving the date at that. As the publishing date would be in June.

> date: 2025-06-07T06:00:00+01:00

Reviewer: @iambami Can we now update this date?

type: Communication
tags:
  - Project Status
  - Community Spotlight
  - Marketing

> cover: 

Reviewer: @iambami Kindly add cover page for this blog post.

authors:
  - name: Atinuke Bami Kayode
    photo: /img/avatars/bami.webp
    link: https://www.linkedin.com/in/atinuke-Bami-kayode-5b838b1b7/
    byline: AsyncAPI Community Marketing Specialist
excerpt: "When Aayush made his first contribution to AsyncAPI, he didn’t expect it would lead to mentorship, leadership, and a thriving open source career."
---

> Open-source communities don’t just grow with code, they grow with people. Every successful project is powered by individuals whose passion, persistence, and commitment light the way for others.

Reviewer 4: ```suggestion
Open-source communities don’t just grow with code, they grow with people. Every successful project is powered by individuals whose passion, persistence, and commitment pave the way for others.
```

In this month’s **Community Spotlight,** we’re thrilled to feature [Aayush Saini](https://www.linkedin.com/in/aayush-saini-0a25931b1/), a shining example of how consistent contributions and a willingness to learn can evolve into leadership. From making his first pull request to mentoring others through Google Summer of Code, Aayush’s story is a powerful reminder of what’s possible when a supportive community and personal drive meet.
In this interview, Aayush shares his journey, challenges, growth, and advice for new contributors stepping into the AsyncAPI world.

## The Interview

**Bami: Nice to meet you, Aayush! I saw the map of your journey, it’s super interesting. Your name was even nominated by [V. Thulisile Sibanda](https://www.linkedin.com/in/v-thulisile-sibanda/) initially. Thanks for making it easy to reach out. Shall we get started?**

> **You didn’t start with Google Summer of Code, right? You started as a regular contributor?**

Reviewer 4: ```suggestion
**Bami: You didn’t start with Google Summer of Code, right? You started as a regular contributor?**
```

Reviewer 2: Did you address this suggestion @iambami?

**Bami: The support system really makes a difference. Was there a specific moment when you realized, "I'm no longer just a contributor, I'm becoming a leader here"?**

***Aayush:** Yes. It happened during the mentorship phase. When I started mentoring others and guiding them through the contribution process, I realized I had internalized a lot of knowledge. It wasn’t just about solving issues anymore, it was about enabling others to succeed too. That shift in mindset changed everything for me.*

**Bami: Awesome. Did you have to learn any new tools, frameworks, or languages along the way?**

***Aayush:** Yes! During the mentorship, I had to learn Jest for testing TypeScript applications. Writing proper test cases was a new skill for me, and it ended up being crucial. I even found more issues related to testing and created a separate project proposal for CLI improvements in Jest afterward.*

**Bami: You’ve moved from learning to teaching, that's real growth! What advice would you give to someone just starting out and feeling intimidated by open-source projects?**

> ***Aayush:** Start small. Even reading documentation, testing features, or fixing typos are contributions. Don't wait to be an expert. Learn as you go. And most importantly, ask questions, a supportive community will always help you.*

Reviewer 5: ```suggestion
***Aayush:** Start small. Even reading documentation, testing features, or fixing typos are contributions. Don't wait to be an expert. Learn as you go. And most importantly, ask questions; a supportive community will always help you.*
```

**Bami: That's beautiful to hear. And what keeps you motivated to keep contributing and advancing within the AsyncAPI community?**

***Aayush:** Honestly, I love this organization. The work is exciting, the community support is amazing, and it's fascinating to see so many companies use AsyncAPI. Beyond that, my goal has been to stabilize the CLI project, onboard more contributors, and help grow the community further.*

**Bami: That’s inspiring. Let’s talk more about your role as a mentor now. You mentioned you became a GSoC mentor after your own mentorship experience. What inspired you to take that step?**

***Aayush:** Yes. I wasn’t a GSoC mentee myself—I went straight from the AsyncAPI Mentorship Program to becoming a GSoC mentor. During my mentorship, I realized that the CLI project still had many improvements to be made. It wasn't possible to address everything within a single program, so I decided to own the CLI project, onboard new folks, and lead it toward more stability and new features.*

**Bami: Amazing. Okay, next one: What advice would you give to a new contributor looking to follow a similar path?**

> ***Aayush:** Okay, the main suggestion I would give is: start slow and be patient. Open source is a different kind of environment, if you don't have patience, it can cause issues. Sometimes we see people who create an issue and then keep pinging maintainers constantly to review it. But you have to understand: everyone has their own timelines, and most contributors are also working full-time jobs. So first, understand the organization and the project properly. Start with the README, set up the codebase successfully, and debug it. Once you can do that, you’ll find multiple issues yourself. Begin with solving the existing "Good First Issues." After that, don't just focus on solving your own issues. Start helping other contributors too,open new issues, suggest enhancements, fix bugs. Your goal should be to contribute to the organization as a whole, not just your own tasks. Keep contributing whenever you have time. That’s what I did. Over time, opportunities will naturally come — like I got the chance to be an SNKP mentee, then a mentor. So yeah, that's the path I suggest for anyone wanting to contribute seriously to SNKP.*

Reviewer 5: ```suggestion
***Aayush:** Okay, the main suggestion I would give is: start slow and be patient. Open source is a different kind of environment; if you don't have patience, it can cause issues. Sometimes we see people who create an issue and then constantly ping maintainers to review it. However, you must understand that everyone has their timeline, and most contributors are also working full-time jobs. First, understand the organization and the project thoroughly. Start with the README, set up the codebase successfully, and debug it. Once you can do that, you’ll find multiple issues yourself. Begin by solving the existing "Good First Issues." After that, don't just focus on solving your issues. Start helping other contributors too, open new issues, suggest enhancements, and fix bugs. Your goal should be to contribute to the organization as a whole, rather than focusing solely on your tasks. Keep contributing whenever you have time. That’s what I did. Over time, opportunities will naturally arise — as I had the chance to be an SNKP mentee, then a mentor. So, yeah, that's the path I suggest for anyone who wants to contribute seriously to SNKP.*
```

**Bami: Great advice. Now, generally, not just during mentorship or maintainership — what are some challenges you faced on your journey, and how did you overcome them?**

> ***Aayush:** In the beginning, it was really hard for me to understand the codebase. I was in a rush, raising PRs without fully understanding the requirements. That's a big mistake. You should always understand what you're doing, what the issue is, and what the expected outcome is. Another challenge was missing or outdated documentation. Sometimes I had to reach out directly to maintainers to understand the workflows. Over time, I contributed to updating documentation myself. That really helped.*

Reviewer 5: ```suggestion
***Aayush:** In the beginning, it was really hard for me to understand the codebase. I was in a rush, raising PRs without fully understanding the requirements. That's a big mistake. You should always understand what you're doing, what the issue is, and what the expected outcome is. Another challenge was missing or outdated documentation. Sometimes I had to reach out directly to maintainers to understand the workflows. Over time, I contributed to updating the documentation myself. That really helped.*
```

**Bami: Got it. So now to a lighter question: What’s one memorable moment or achievement in your journey?**

> ***Aayush:** One of my best moments was completing an advanced bounty issue for the AsyncAPI Generator project. It was a really tough task, but with help from Lucas, I managed to complete it on time. That felt amazing.*

Reviewer 4: ```suggestion
***Aayush:** One of my best moments was completing an advanced bounty issue for the AsyncAPI Generator project. It was a really tough task, but with help from Lukasz, I managed to complete it on time. That felt amazing.*
```

**Bami: Nice! If you were to give shout-outs, who would you mention?**

> ***Aayush:** Honestly, I would shout out the whole AsyncAPI community. Everyone is doing great work. But if I have to mention names, then definitely Sovik, who was my first mentor, and also Ashish, Sameer, Lucas Fran, and many others, including you.*

Reviewer 4: ```suggestion
***Aayush:** Honestly, I would shout out the whole AsyncAPI community. Everyone is doing great work. But if I have to mention names, then definitely Sovik, who was my first mentor, and also Ashish, Sameer, Lukasz, Fran, and many others, including you.*
```

**Bami: That's sweet. How do you envision the AsyncAPI community evolving in the future, based on your experience?**

***Aayush:** I see a very bright future. There’s still a lot of work to do, and we need more contributors. Programs like the mentorship initiative will help onboard new people and strengthen the community. We also have strong community and company support now, so I’m very optimistic.*

**Bami: Last serious question: As a maintainer, mentor, and TSC member, what would you like to accomplish next?**

***Aayush:** I want to take ownership of the Studio project. I'm currently learning frontend technologies and also working on a new Java + Quarkus template project for AsyncAPI. Besides that, I'm involved in setting up systems like the GitHub voting automation for TSC elections. Lots of fun projects ahead!*

**Bami: Nice! Speaking of the TSC voting automation, that was a bounty project, right? How was your experience working on it?**

> ***Aayush:** Yes, it was a bounty. It was very interesting because it revolved around creating a GitHub workflow automation. At the time, I wasn't very familiar with GitHub Actions, so there was a learning curve. Testing was a big challenge because you can’t directly test GitHub Actions just by pushing a PR, I had to test everything from my private repo. In total, I made about 400 commits just to test and ensure everything worked properly before pushing to production!*

Reviewer 5: ```suggestion
***Aayush:** Yes, it was a bounty. It was very interesting because it revolved around creating a GitHub workflow automation. At the time, I wasn't very familiar with GitHub Actions, so there was a learning curve. Testing was a big challenge because you can’t directly test GitHub Actions just by pushing a PR I had to test everything from my private repo. In total, I made about 400 commits just to test and ensure everything worked properly before pushing to production!*
```

**Bami: 400 commits? Wow, that’s impressive! That could be a blog post on its own. Okay, now let’s move into the fun part, a quick “Have You Ever” game.**

## Have You Ever - AsyncAPI Bingo

**Bami: Have you ever spent an entire night fixing a bug?**

> ***Aayush:** No, not the whole night, maybe 5-6 hours but not overnight.*

Reviewer 5: ```suggestion
***Aayush:** No, not the whole night, maybe 5-6 hours, but not overnight.*
```

**Bami: Have you ever created a PR you were super proud of?**
***Aayush:** Yes, definitely.*

**Bami: Have you ever accidentally broken something important while coding?**
***Aayush:** Yes, during my early days at AsyncAPI, I broke a feature, and a stakeholder noticed it within five minutes! Thankfully, Sameer quickly fixed it.*

**Bami: Have you ever helped a new contributor find their way in the community?**
***Aayush:** Yes, many times.*

**Bami: Have you ever had a "Eureka!" moment while coding?**
***Aayush:** Yes, while working on a command in the CLI project, it helped me discover and solve many hidden issues.*

**Bami: Have you ever joined a meeting in your pajamas?**
***Aayush:** Haha, yes!*

**Bami: Have you ever been surprised by how fast someone reviewed your PR?**

> ***Aayush:** Yes, [Lukasz Gornicki](https://www.linkedin.com/in/lukasz-gornicki-a621914/) once reviewed my PR very quickly and I was amazed.*

Reviewer 5: ```suggestion
***Aayush:** Yes, [Lukasz Gornicki](https://www.linkedin.com/in/lukasz-gornicki-a621914/) once reviewed my PR very quickly, and I was amazed.*
```
